build-system/tasks/visual-diff/index.js

```diff
@@ -535,24 +566,31 @@ async function snapshotWebpages(percy, browser, webpages) {
           .catch(async testError => {
             log('travis', colors.red('●'));
             addTestError(testErrors, name, 'Unknown failure in test page',
-                testError, /* fatal */ true);
+                testError, consoleMessages);
 
+            let htmlSnapshot;
+            try {
+              htmlSnapshot = await page.content();
+            } catch (e) {
+              htmlSnapshot = e.message;
+            }
             await page.setContent(
                 SNAPSHOT_ERROR_SNIPPET
                     .replace('__TEST_NAME__', name)
-                    .replace('__TEST_ERROR__', testError));
+                    .replace('__TEST_ERROR__', testError)
+                    .replace('__HTML_SNAPSHOT__', escapeHtml(htmlSnapshot)));
```

Comment:
For debugging ease, maybe also provide the path of the example page used for the test?